### ISSUE TITLE
PWG ls markup 

### ISSUE THREAD

**Issue by funderburkjim** [sanskrit-lexicon/PWG#51]
I am aiming to resolve all the literary source links in PWG for which there is a link target.
Currently this includes RV, AV, MBH, HARIV  (and probably soon) R(amayana).

This note documents the current skirmishes in this battle.

The first intermediate objectives:
* to make visible all the ls sources
  * Example: `<ls>AV. 1, 2, 3. HARIV. 12345.</ls>` -> `<ls>AV. 1, 2, 3.</ls> <ls>HARIV. 12345.</ls>`
* to remove unneeded text from ls references.  
  * Example: `(<ls>nach</ls> <ls>HAUGHTON).</ls>`  -> `(nach <ls>HAUGHTON).</ls>`
* to join multi-word references.
  * Example: `<ls>R.</ls> <ls>GORR.`  -> `<ls>R. GORR.`

Also, a few changes were made unrelated to 'ls'. e.g. `{#(X)#}` -> `({#X#})`  (7000).

**Comment by funderburkjim** [comment]
## first phase done.

Notes describing the changes is present in [pwg_ls2/mbh1 folder](https://github.com/sanskrit-lexicon/PWG/tree/master/pwg_ls2/mbh1).
About 28000 lines of pwg.txt changed, or about 2.5% of the lines.

The changes were done in six steps, with the detailed changes of each step in change_01.txt through change_06.txt.

In addition, a summary of the current frequency count of ls markup was prepared
[s summary [before](https://github.com/sanskrit-lexicon/PWG/blob/master/pwg_ls2/mbh1/lsextract_pwg_00.txt) and [after](https://github.com/sanskrit-lexicon/PWG/blob/master/pwg_ls2/mbh1/lsextract_pwg_06.txt),

**Comment by gasyoun** [comment]
> Currently this includes RV, AV, MBH, HARIV (and probably soon) R(amayana).

Hurray, that was top 3 of my whishlist and now it's become real.

> [s summary [before](https://github.com/sanskrit-lexicon/PWG/blob/master/pwg_ls2/mbh1/lsextract_pwg_00.txt) and [after](https://github.com/sanskrit-lexicon/PWG/blob/master/pwg_ls2/mbh1/lsextract_pwg_06.txt),

Thanks, its was badly wanted, an update.

> 20121	ŚKDR.	ŚABDAKALPADRUMA (GILD. Bibl. 371).

@funderburkjim what would be the way to link to a book we allready have? What help would be required from my volunteers?

> 19829	P.	PĀṆINI'S acht Bücher grammatischer Regel

Done in MW, so can be redone easily?

> 19005	BHĀG. P.	BHĀGAVATAPURĀṆA, nach Anführungen im VP.

Have no idea about it. How about @Andhrabharati ? @SergeA ?

> 16030	H.	HEMACANDRA'S ABHIDHĀNACINTĀMAṆI, ein sys

@drdhaval2785 where is your Hemacandra?

>15633	HARIV.	HARIVAṂŚA im 4ten Bande des MBH.&#13;&#1

Long live Kate

> 14914	KATHĀS.	KATHĀSARITSĀGARA, ed. BROCKHAUS (GILD. B

I'm working on publishing the Russian traslation once again, but it is a different edition.

> 14467	AK.	AMARAKOṢA nach der Ausgabe von COLEBROOK

I have the book, it's a gem. Guess it does not differ in the Sanskrit part from the later editions @drdhaval2785 ?

> 12975	MED.	MEDINĪKOṢA, ed. Calc. (GILD. Bibl. 258).

Waiting for @drdhaval2785

>12497	Verz. d. Oxf. H.	? 

Nothing to be done here?

>12384	M.	MANU'S Gesetzbuch in der Ausg. von LOISE

I believe we will have to leave it for later.

10598	ŚAT. BR.	The ŚATAPATHABRĀHMAṆA in the Mādhyandina
09776	H. an.	HEMACANDRA'S ANEKĀRTHASAM̃GRAHA, ed. Cal
09687	SUŚR.	SUŚRUTA, ed. Calc. (GILD. Bibl. 367); ci
09360	Spr.	Indische Sprüche. Sanskrit und deutsch h
09281	AV.	ATHARVAVEDASAM̃HITĀ, herausg. von R. ROT
08361	TRIK.	TRIKĀṆḌAŚEṢA, ed. Calc. (GILD. Bibl. 258
07986	VARĀH. BṚH. S.	VARĀHAMIHIRA'S BṚHATSAM̃HITĀ.
07733	RAGH.	RAGHUVAṂŚA, ed. STENZLER (GILD. Bibl. 13
07309	Spr. (II)	Indische Sprüche. Sanskrit und Deutsch. 
07118	RĀJA-TAR.	RĀJATARAṄGIṆĪ, ed. TROYER (GILD. Bibl. 1
07065	PAÑCAT.	PAÑCATANTRA. Pantschatantrum sive quinqu
06873	Sch.	Scholiast.
06648	RĀJAN.	RĀJANIRGHAṆṬA, ein medicinisches Wörterb
06008	Ind. St.	WEBER'S Indische Studien.
05196	MĀRK. P.	MĀRKĀṆḌEYAPURĀṆA in der Bibliotheca indi
05107	HALĀY.	HALĀYUDHA, ein Lexicograph; nach Anführu
05070	KĀTY. ŚR.	KĀTYĀYANA'S ŚRAUTASŪTRĀṆI. 26 Adhyāya. H
05058	VS.	VĀJASANEYISAM̃HITĀ. The VĀYASANEYI-SANHI
05036	VOP.	VOPADEVA'S Grammatik, nach BÖHTLINGK'S A
04926	ŚĀK.	ŚĀKUNTALA, das Drama in der Ausgabe von 
04593	TS.	TAITTIRĪYASAM̃HITĀ. 7 Kāṇḍa, eingetheilt
04023	R. GORR.	RĀMĀYAṆA, translation by Gaspare Gorresi
03779	AIT. BR.	AITAREYABRĀHMAṆA. Citirt nach den 8 Pañc
03649	Verz. d. B. H.	WEBER'S Verzeichniss der Berliner Sanskr
03621	VP.	VIṢṆUPURĀṆA. Es wird die WILSON'sche Ueb
03543	YĀJÑ.	YĀJÑAVALKYA'S Gesetzbuch. Sanskrit und D
03519	SĀH. D.	SĀHITYADARPAṆA, auf den 10 ersten Bogen 
03128	ŚABDAR.	ŚABDARATNĀVALĪ, ein Wörterbuch; nach Anf
03049	Schol.	Scholiast.
03027	NIR.	YĀSKA'S NIRUKTA sammt den NIGHAṆṬAVA'S h
02981	N.	NALOPĀKHYĀNA in BÖHTLINGK'S Chrestomathi
02976	HIT.	HITOPADEŚA, ed. SCHLEGEL und LASSEN (GIL

**Comment by Andhrabharati** [comment]
> > 19005	BHĀG. P.	BHĀGAVATAPURĀṆA, nach Anführungen im VP.
> 
> Have no idea about it. How about @Andhrabharati ? @SergeA ?

When followed by a number, it denoted the Burnouf's edition, which got published in 3 vol.s (Vol-1: Skandhas 1-3; Vol-2: Skandhas 4-6; Vol-3: Skandhas 7-9), containing both Sanskrit verses and French translation.
[After a long gap (~30 years), other skandhas were 'done' by other scholars, but they are not to be used as citations!]

**Comment by Andhrabharati** [comment]
> > 14914	KATHĀS.	KATHĀSARITSĀGARA, ed. BROCKHAUS (GILD. B
> 
> I'm working on publishing the Russian traslation once again, but it is a different edition.

Kathasaritsagara of Brockhaus was published in 3 parts (Part-1: lambakas 1-5, Part-2: lambakas 6-8 & Part-3: lambakas 9-18) in 1839.

**Comment by Andhrabharati** [comment]
> > 12384	M.	MANU'S Gesetzbuch in der Ausg. von LOISE
> 
> I believe we will have to leave it for later.

Laws of Manu by Deslongchamps was published in 1830, and it is a widely cited work. Why leave it?

**Comment by Andhrabharati** [comment]
> > 12497	Verz. d. Oxf. H.	?
> 
> Nothing to be done here?

I had given the link to this widely cited work sometime back, and indeed it could be indexed for linking now.

> 10598 ŚAT. BR. The ŚATAPATHABRĀHMAṆA in the Mādhyandina

The Śatapatha Brāhmaṇa by A. Weber is a major work after the Vedas, is too popular a work to be left out.

**This covers the works with 10k+ citations in PWG.**
`--------------`
Can we expect similar counts being listed by @funderburkjim in pwk as well?

**Comment by Andhrabharati** [comment]
The second phase could cover the following works (5k-10k):

- Suśruta
- (Ind.) Spr.
- Bṛhatsam̃hitā
- Raghuvaṃśa
- Rājataraṅgiṇī
- Pañcatantra
- Kātyāyana's Śrautasūtrāṇi
- Vājasaneyisam̃hitā
- and finally, Boethlingk's Vopadeva

**Comment by gasyoun** [comment]
> Can we expect similar counts being listed by @funderburkjim in pwk as well?

Yes, please.

>The second phase could cover the following works (5k-10k):

(Ind.) Spr. is already uploaded, but not linked to?

`Raghuvaṃśa` and `Pañcatantra` I would keep higher in the list.

@Andhrabharati if you would give the PDFs for @katerusse I would ask her to go for `Kathasaritsagara of Brockhaus`

**Comment by Andhrabharati** [comment]
Just look for "katha sarit sagara Brockhaus" in Google books; two books with years 1839 (vol.1) and 1862 (vol.2: 1862 & vol.3: 1866) can be downloaded from there.

You can get these from archive.org as well.

**Comment by Andhrabharati** [comment]
> > Can we expect similar counts being listed by @funderburkjim in pwk as well?
> 
> Yes, please.

I just recalled that @funderburkjim has recently posted the pwk ls lists [summary & detail] (in November last); but probably he could update the same now, which might just change the MBh. & Hariv. numbers.

**Comment by gasyoun** [comment]
> Just look for "katha sarit sagara Brockhaus" in Google books; two books with years 1839 (vol.1) and 1862 (vol.2: 1862 & vol.3: 1866) can be downloaded from there.

https://www.google.ru/books/edition/Katha_Sarit_Sagara_Die_M%C3%A4hrchensammlung/MLpUAAAAcAAJ?hl=ru&gbpv=1&dq=katha+sarit+sagara+Brockhaus&printsec=frontcover @KateRusse  - do you understand the next step?

**Comment by funderburkjim** [comment]
## MBH additional markup

This work described in [pwg_ls2/mbh2 directory](https://github.com/sanskrit-lexicon/PWG/tree/master/pwg_ls2/mbh2).

The change_01.txt shows the sequence of changes.  It is large.

A typical change would be of form: `<ls>X, Y.</ls> -> <ls n="MBH.">X, Y.</ls>`  Where X and Y are digit sequences.  At this point in the development of pwg.txt, there are about 48000 items marked as `<ls>X...`  (X a number).  But not all of these pertain to MBH.  So we add the MBH markup only if we determine that the number markup is a continuation of previous MBH markup.

This determinationis made by seeing if the previous non-number ls is an explicit MBH.

For example, under L=41, aMsay pwg.txt has
```
<ls>MBH. 1, 197.</ls> {#vyaMsayAmAsa taM tasya prahAram#} 
<ls>3, 11728.</ls>
```
so it is appropriate to change the markup `<ls n="MBH.">3, 11728.</ls>`.

But in a case such as
```
<ls>RAGH. 6, 12.</ls> {#pUraBivyaktamuKaprasAdA#} 
<ls>16, 23.</ls> {#anaBivyaktAScandrikAyAM dIpikAH punaruktAH#} 
```
it is not appropriate to add MBH markup `<ls n="MBH.">16, 23.</ls>`,  rather the correct markup
would be `<ls n="RAGH.">16, 23.</ls>`

Of course, there are several other variations to this basic example.

The summary of the markup changes can be seen in this comparison:
```
685196	ALL	As of 2022-02-13
48003	NUMBER	ls starts with number
02157	UNKNOWN	ls is unknown
55834	MBH.	MAHĀBHĀRATA, ed. Calc. (GILD. Bibl. 93).

688014	ALL	As of 2022-02-19
39759	NUMBER	ls starts with number
02163	UNKNOWN	ls is unknown
66809	MBH.	MAHĀBHĀRATA, ed. Calc. (GILD. Bibl. 93).
```

**Comment by gasyoun** [comment]
> The summary of the markup changes can be seen in this comparison:

A 10k increase, amazing work as usual.

**Comment by funderburkjim** [comment]
## Additional RV1 markup

Another additional 1000+ Rigveda links marked for link target.
Work done at https://github.com/sanskrit-lexicon/PWG/tree/master/pwg_ls2/rv1.

There is a rather large (500) number of RV references with 2 numbers.   
See [abnormal_rv_01.txt](https://github.com/sanskrit-lexicon/PWG/blob/master/pwg_ls2/rv1/abnormal_rv_01.txt) or the slightly more elaborate [change_abnormal.txt](https://github.com/sanskrit-lexicon/PWG/blob/master/pwg_ls2/rv1/change_abnormal.txt).

At least some of these are references to a particular hymn, rather than to a particular verse of a particular hymn.

**Comment by funderburkjim** [comment]
## Additional AV markup

These changes in markup expose additional active links to the Atharva Veda verses.

Some examples:
```
; <L>7<pc>1-0003<k1>aMSa<k2>a/MSa<h>1
;  expose link to the 11, 17, 2 reference
161 old <ls>AV. 6, 2, 5. 11, 17, 2.</ls> Name des 6ten <is>Āditya</is>, 
161 new <ls>AV. 6, 2, 5.</ls> <ls n="AV.">11, 17, 2.</ls> Name des 6ten <is>Āditya</is>, 

; <L>30796<pc>3-0417<k1>trayodaSa
; preceding line mentions AV:
;  <ls>AV. 5, 6, 4.</ls> {#tri\MSada^NgaM trayoda\SaM mAsa^m#} 
; 13, 3, 8 inferred to be verse in AV
305038 old <ls>13, 3, 8.</ls> <ls>ŚAT. BR. 4, 3, 1, 5. 6, 2, 2, 29.</ls> {#yAni paYjatriMSatsa trayodaSo mAsaH#} 
; and markup is added so link to 13,3,8 is recognized by displays
305038 new <ls n="AV.">13, 3, 8.</ls> <ls>ŚAT. BR. 4, 3, 1, 5. 6, 2, 2, 29.</ls> {#yAni paYjatriMSatsa trayodaSo mAsaH#} 
```

The markup work was done in folder [pwg_ls2/av](https://github.com/sanskrit-lexicon/PWG/tree/master/pwg_ls2/av).

The revised count of AV links rose from 9283 to 16149.  
(compare latest [lsextract_pwg.txt](https://github.com/sanskrit-lexicon/PWG/blob/master/pwg_ls2/av/lsextract_pwg.txt) to [previous lsextract_pwg.txt](https://github.com/sanskrit-lexicon/PWG/blob/master/pwg_ls2/rv1/lsextract_pwg.txt).)

**Comment by funderburkjim** [comment]
## 11, 17, 2 not found
In the above example, under aMSa,  the AV link to 11, 17, 2  is not found! We get a 404 error from
https://sanskrit-lexicon.github.io/avlinks/avhymns/av11.017.html.

Don't know the cause of this.

**Comment by Andhrabharati** [comment]
> In the above example, under aMSa, the AV link to 11, 17, 2 is not found! We get a 404 error from
> https://sanskrit-lexicon.github.io/avlinks/avhymns/av11.017.html.

Not only this AV. 11,17,2; but the AV. 6,2,5 also is NOT at the link.

> Don't know the cause of this.

The reason is very plain and simple-- the VN text is to be read and applied appropriately to the main pages' text.

In the above case,
`<ls>AV. 6,2,5. 11,17,2.</ls>` at `<L>7` is to be corrected with
`<ls>Z. 3 lies 6,4,2. 11,6,2 st. 6,2,5. 11,17,2.</ls>` as mentioned at `<L>62405`.

Thus the citations become 
`<ls>AV. 6,4,2. 11,6,2.</ls>` which are present in the resp. link pages--

avś_06.004.02.
अं॒शो भ॒गो व॒रुणो मित्रो॒ अर्यमा॒दितिः पा॒न्तु मरु॒तः ।
अ॒प त॒स्य द्वे॒षो गमेद् अभिह्रु॒तो यावयच् छ॒त्रुम् अ॒न्तितम् ॥२॥

avś_11.006.02.
ब्रूमो॒ रा॒जानं व॒रुणं मित्रं॒ वि॒ष्णुम् अ॒थो भ॒गम् ।
अं॒शं वि॒वस्वन्तं ब्रूमस् ते॒ नो मुञ्चन्त्व् अं॒हसः ॥२॥

**There are very many such link targets that need corrections from the VN pages.**

Does this sound a good reason for the CDSL team, to "applying" the VN text to the main text (i.e. integrating appropriately; not just appending somewhere beneath, as is being done now)?

**Comment by gasyoun** [comment]
> atleast the 'lies ... ... st. ... ...' portions?

Sounds like a big piece of work.

**Comment by Andhrabharati** [comment]
There are just about 1200 of 'lies' strings in the whole of PWG text.

Not a big thing at all, if decided to be 'done'.

**Comment by funderburkjim** [comment]
Thanks for the research on `<ls>AV. 6,2,5. 11,17,2.</ls>`, exposing the role of the VN material.

**Comment by funderburkjim** [comment]
## Additional P markup

These changes in markup expose additional active links to Panini; the work
is similar to that for AV.

An example
```
; <L>14932<pc>2-0067<k1>kapIvant
OLD:
<ls>P. 6, 3, 120. 8, 2, 11,</ls>
NEW:
<ls>P. 6, 3, 120.</ls> <ls n="P.">8, 2, 11</ls>,

Now the 8,2,11 link is exposed and appears in displays.
```

The markup work was done in folder [pwg_ls2/p](https://github.com/sanskrit-lexicon/PWG/tree/master/pwg_ls2/p).

The revised count of P links rose from 19829 to 24914
(compare latest [lsextract_pwg.txt](https://github.com/sanskrit-lexicon/PWG/blob/master/pwg_ls2/p/lsextract_pwg.txt) to [previous lsextract_pwg.txt](https://github.com/sanskrit-lexicon/PWG/blob/master/pwg_ls2/rv1/lsextract_pwg.txt).)

**Comment by gasyoun** [comment]
> The revised count of P links rose from 19829 to 24914

Big fish.

**Comment by Andhrabharati** [comment]
> The revised count of P links rose from 19829 to 24914

@funderburkjim 

Just like to bring to your notice that not all "P. xxx" citations refer to Panini's work **as is**.

Many a times they are to its associated works [commentaries (or improvements) on it] like Kātyāyana’s Vārttika, Patanjali’s Mahābhāṣya, Jayāditya’s Kāśikā (**referred mostly as Sch. in this PWG, and hence in MW**), or Bhattoji's Siddhānta Kaumudi (SK).

See for e.g.
`<L>16<pc>1-0004<k1>अंशहर<k2>अंशहर꣫ `
{#अंशहर꣫#}¦ (1. {#अंश + हर#}) <lex>adj.</lex> {%einen Erbschaftsantheil empfangend, erbend%}, `<ls>P. 3,2,9, Sch.</ls>` `<ls>YĀJÑ. 2,132. 133.</ls>`
`<LEND>`

This "P. xxx" is presently in two ls parts in CDSL text-- `<ls>P. 3, 2, 9,</ls> <ls>Sch.</ls> ` and is linked to P. 3.2.9 "हरतेरनुद्यमनेऽच्", which not even remotely has any indication of the word अंशहर.

It is the associated Kāśikā that has the अंशहरः word--
"हरतेर्धातोरनुद्यमने वर्तमानात् कर्मणि उपपदे अच् प्रत्ययो भवति। अणोऽपवादः। उद्यमनमुत्क्षेपणम्। **अंशं हरति अंशहरः**। रिक्थहरः। अनुद्यमने इति किम्? भारहारः। अच्प्रकरणे शक्तिलाङ्गलाङ्कुशयष्टितोमरघटघटीधनुष्षु ग्रहेरुपसङ्ख्यानम्। शक्तिग्रहः। लाङ्गलग्रहः। अङ्कुशग्रहः। यष्टिग्रहः। तोमरग्रहः। घटग्रहः। घटीग्रहः। धनुर्ग्रहः। सूत्रे च धार्यर्थे। सूत्रग्रहः। सूत्रं धारयति इत्यर्थः। सूत्रग्राहः एव अन्यः।"

**In summary, we should not split these as separate parts but have them as single entity (as done in my version); and link directly to the resp. work as far as possible.** 
; ------------------------
P.S. A small clue is that, the comma in these citations portion is to be taken as a continuation mark, not a separator (like . or ;).

**Comment by funderburkjim** [comment]
In the example (P. 3,2,9)   The link https://ashtadhyayi.com/sutraani/3/2/9  also shows the Kāśikā . So in this example at least,  the link target is relevant -- I see no problem here.

**Comment by gasyoun** [comment]
> So in this example at least, the link target is relevant -- I see no problem here.

@Andhrabharati your say?

**Comment by Andhrabharati** [comment]
I chose P. just as an example for the point 'of clubbing commentary with the actual main source'.

It is just incidental here that astadhyayi.com is used as the link target, which has almost all the associated works together at a single place; and as such @funderburkjim 'felt' no 'relevance' of my point.

When other works are linked, the significance would be known. The works list is quite vast, starting from SAY on Vedas &c., KULL on Manu, MIT on Yaj~n, NILAK on MBH, MALL on various kAvyas. ... ...; If only the main work is linked in these cases, the actual intended commentary text would not be there. So the proper link text/source is to be used in all such cases.

Also the point could be extended to another variety, "citations within another work"; these could mostly be taken as "xxx im (or in) SKDR", being the cited reference as in SKDR. Seen that other editions of those actual works sometimes have a different text that that is in SKDR.

**Comment by Andhrabharati** [comment]
And my actual point has another aspect, to combine both ls parts together.

As can be seen even in this example, the Sch. part (separately marked) goes anywhere (which actually is the intended matter); though the link to the first P. xxx part nevertheless contained the Sch. text somewhere underneath.

**Comment by funderburkjim** [comment]
## Additional R. markup
This work described in #57.

**Comment by funderburkjim** [comment]
@Andhrabharati  In re-reading your comments above,

First, the 'P.' example is certainly a special case, in that the link target includes access to the commentaries.  As you described, an interested and informed reader can get to the desiderata  from the link given to the sutra.  This seems satisfactory to me.  While it might be possible to devise a link to a particular commentary for a given verse, I'm not sure that this added specificity would be of great use.
But I'm open to considering it further.

As an exemplar for the other references, let's take sAyaRa commentary on Rg veda.
In PWG there are  currently 180+ references with the form `<ls>SĀY.</ls> zu <ls>ṚV. ...`,
The first one is:
```
<L>296<pc>1-0023<k1>aKKala<k2>aKKala
{#aKKala#}¦ Interj. der Freude und Ueberraschung: {#aKKalaSabdaM kftvA#} 
<ls>SĀY.</ls> zu <ls>ṚV. 7, 103, 3.</ls>
<LEND>
```
The  ṚV. reference is linked to the verse, and the aKKala word does occur.  So this link has at least some
utility.
We currently have no link target for sAyaRa's commentary.   But if we did, and if we could determine
exactly where in the sAyaRa commentary this word in this verse of RV is located in the link target, then
we could add the required parameters:  `<ls n="page x of sayana rv">SĀY.</ls>`.   

Thus, it seems to me that in this case, the current markup is both satisfactory for the current state of our link targets, and upgradeable.  

Practically, there would be  several steps involved to upgrade the markup for such a commentary:
* get link target.  Where are pdfs of the commentaries? 
* develop page reference scheme for link target (easy)
* develop rule to determine link page reference for each specific commentary instance. 
  and add the markup to pwg.txt. (probably hard)
* revise basicadjust.php to make use of the commentator link. (probably easy)

@Andhrabharati Do you see flaws in this suggested approach?

In terms of priorities,  I might prefer to develop link targets for some of the other more common ls references of pwg.  But a 'complete' markup of pwg would include links to commentaries.

**Comment by Andhrabharati** [comment]
There are no flaws in your approach, @funderburkjim ; you're quite practical in dealing with any matter (when you take it up!).

What I was saying is to get/show the actual 'quoted text' as far as possible, not to just give an indication of the 'entry word'.

Here is the screenshot for the above RV citation with actual wording used in PWG--

![image](https://user-images.githubusercontent.com/75209130/175213854-bf14fa75-ef86-4bd9-8390-63e0d1e0260d.png)

**As you had pointed it, it is just the matter of finding the 'right' (PDF) source to link-up the citation.**

**Comment by Andhrabharati** [comment]
> an interested and informed reader can get to the desiderata

Whether a link is provided or not, this is TRUE in every sense.
An 'interested and informed reader' would get what is 'needed' for him, SOMEHOW.

**Comment by gasyoun** [comment]
> possible to devise a link to a particular commentary for a given verse, I'm not sure that this added specificity would be of great use.

Technically doable?

**Comment by Andhrabharati** [comment]
Jim has already stated the steps above; undoubtedly 'do-able', if the sources are identified/traced.

**Comment by gasyoun** [comment]
> prefer to develop link targets for some of the other more common ls references of pwg. 

Agree

>  a 'complete' markup of pwg would include links to commentaries.

Guess even @Andhrabharati agrees it would be too much for now

**Comment by Andhrabharati** [comment]
I am a kind of 'a perfectionist'!!

**Comment by gasyoun** [comment]
> [After a long gap (~30 years), other skandhas were 'done' by other scholars, but probably they are not to be used as citations here!]

@Andhrabharati seen the scan?

**Comment by Andhrabharati** [comment]
You mean scans of the parts done by others?
YES.

The reason behind my above statement, is not that the data is bad, but that it is not what PWG or pwk has seen.
These portions came out much later.

**Comment by Andhrabharati** [comment]
Here are some snippets from the **Bomb. ed. of Bhag. P.**--

[Part-1 containing Sk.1-7]
![image](https://github.com/sanskrit-lexicon/PWG/assets/75209130/5da952c9-3cab-447e-89f8-34b46b792b2f)
![image](https://github.com/sanskrit-lexicon/PWG/assets/75209130/996151bb-3e78-4cb7-a7ec-d0a9b7dafbd3)

![image](https://github.com/sanskrit-lexicon/PWG/assets/75209130/aea4fd89-ca6e-4015-8fab-057d4362d95d)
![image](https://github.com/sanskrit-lexicon/PWG/assets/75209130/de0563c1-cb7f-4177-8c25-cd4b375f537d)

![image](https://github.com/sanskrit-lexicon/PWG/assets/75209130/e0586d5d-6e04-46e4-be58-7e008967d238)
![image](https://github.com/sanskrit-lexicon/PWG/assets/75209130/2ad7720e-e89a-4e3b-a4dc-071f7c75f9bf)

![image](https://github.com/sanskrit-lexicon/PWG/assets/75209130/ff07a9ed-39db-46fe-83a7-fe7de1db1ead)
![image](https://github.com/sanskrit-lexicon/PWG/assets/75209130/4a9920a8-72d1-4fb3-a179-d1922b9b0de8)

![image](https://github.com/sanskrit-lexicon/PWG/assets/75209130/8c04a5eb-516b-47a9-be55-acc743f782e1)
![image](https://github.com/sanskrit-lexicon/PWG/assets/75209130/2896d4f5-2628-444d-b4f8-5f5c33899e2f)

![image](https://github.com/sanskrit-lexicon/PWG/assets/75209130/b094edfb-b281-4778-9128-e9c1accf0244)
![image](https://github.com/sanskrit-lexicon/PWG/assets/75209130/a44760d2-8078-42bc-bcf0-cb23150cb470)

![image](https://github.com/sanskrit-lexicon/PWG/assets/75209130/a28da4a8-268c-4ecc-b270-4c90cf76d87b)
![image](https://github.com/sanskrit-lexicon/PWG/assets/75209130/b016562c-78ac-4c9b-99ce-5f10499ba40c)

[Part-2 containing Sk.8-12]
![image](https://github.com/sanskrit-lexicon/PWG/assets/75209130/06ac36d5-49a0-45c8-bfa0-53e7e3573e30)
![image](https://github.com/sanskrit-lexicon/PWG/assets/75209130/9b905f24-b4db-4be3-8f5a-aa4b4264630b)

![image](https://github.com/sanskrit-lexicon/PWG/assets/75209130/f8998304-b712-4726-9192-57826fa0fda2)
![image](https://github.com/sanskrit-lexicon/PWG/assets/75209130/e5337a91-777b-4d5f-92ce-a1b6f0be41c5)

![image](https://github.com/sanskrit-lexicon/PWG/assets/75209130/205d6018-8f48-43c3-9e13-3a3375d25d0f)
![image](https://github.com/sanskrit-lexicon/PWG/assets/75209130/565fb735-5cff-4962-8fc4-7f8b71a8e597)
![image](https://github.com/sanskrit-lexicon/PWG/assets/75209130/5ba7e48d-c4ef-4bde-b1d8-12be10ee1d2b)
![image](https://github.com/sanskrit-lexicon/PWG/assets/75209130/5553a446-0e23-48b1-9719-ae2744b1d150)

![image](https://github.com/sanskrit-lexicon/PWG/assets/75209130/75136349-89da-4986-89ad-8cebd3abea64)
![image](https://github.com/sanskrit-lexicon/PWG/assets/75209130/198cd9c8-089d-4e7f-9a58-6e358c934122)

![image](https://github.com/sanskrit-lexicon/PWG/assets/75209130/82b22238-4af3-416a-bdb3-a9c328d6d492)
![image](https://github.com/sanskrit-lexicon/PWG/assets/75209130/db5e74de-e58e-412e-8f81-11cdfa5747e4)

Transcripted text:
यथाध्वनीनः स्थपुटं प्रयान्पथं चिराय नैजं प्रतिपद्यते क्लिशन् ॥
तथाऽविशुद्धं प्रतिपद्य पुस्तकं बुधोधिगच्छत्यधिगम्यमास्पदं ॥ १ ॥
अनो बुधैः सूक्ष्मदृशाविधेया सा पौस्तकी शोधनिकाऽतियत्नात् ॥
साहाय्यकृद्दत्तविवेक दृग्वचोमनोभिरामाऽक्षरमालिकांचिता ॥ २ ॥
तत्प्रस्तुतं भागवतीयपुस्तकं निरीक्षमाणौ कृतलक्षणान्स्तुवः ॥
क्षेत्रं करोपाभिधनाशिकस्थ गोविंद सद्वैद्यतनूभवोऽन्यः ॥ ३ ॥
रेवदंडापुरीवृत्तिलब्धधर्माधिकारवान् ॥
हरिजोत्र महादेवः शोधं चक्रे यथामति ॥ ४ ॥
कृष्णभूगणपत्याख्य मुद्रायंत्रालयेऽमले ॥
तत्तनूभवकान्होबाभिधेन विदुषां मुदे ||
द्विदिग्गजाद्रिकुमिते (१७८१) रौद्राब्दे शालिवाहके ॥
मार्गे पुण्येऽग्रहायण्यां मुद्रितं मुद्रिकाक्षरैः ॥
[1781  S.S. = 1859 A.D.]
**From this one may know that most (if not all) of the 'famed' early Bomb. ed. works were from the Ganapat Krishnaji Press, Bombay.**

![image](https://github.com/sanskrit-lexicon/PWG/assets/75209130/aeb3f8da-075f-4fb2-a9a0-ef605cfe2fd0)


![image](https://github.com/sanskrit-lexicon/PWG/assets/75209130/176392d0-b6e6-4d57-b25c-18dd2ca1c41e)
![image](https://github.com/sanskrit-lexicon/PWG/assets/75209130/7f0563a3-f83c-40a7-ba20-4c73ee3efaa5)

**Comment by Andhrabharati** [comment]
As I had mentioned before, this **Bh. P.** also came with the Śrīdhara's comm. (Bhāvārthadīpikā); just like the way the other big works **R.** (with Tilaka comm.) and **MBh.** (with Nīlakaṇṭha’s comm.).

**Comment by drdhaval2785** [comment]
I think that ls markup is now achieved. Whatever remains to be linked, can be linked in separate issues.
